packages/api/src/modules/providers/fonjep/repositories/fonjep.subvention.repository.ts

```diff
@@ -37,6 +42,31 @@ export class FonjepSubventionRepository extends MigrationRepository<FonjepSubven
     async rename(name: string) {
         return this.collection.rename(name);
     }
+
+    useTemporyCollection(active) {
+        this.tmpCollectionEnabled = active;
+    }
+
+    async applyTemporyCollection() {
```

Comment:
Je comprend pas cette méthode. Le but c'est de dupliquer la base avant process et la drop seulement si la commande a aboutie ?

Je comprend pas les deux `.rename` à la suite qui ne copie pas vraiment la collection initiale.

J'ai l'impression que `await this.db.collection(this.collectionName + "-OLD").drop();`ne fera jamais rien car on aura déjà exécuté `await this.collection.rename(this.collectionName);` et aucune collection ne s'appelera `this.collectionName + "-OLD"`

Reply:
En gros le principe c'est qu'on à remplie avec les nouvelle données la collection tmp, ici on veux switcher entre la collection original et celle tmp, donc on renome l'original en OLD ensuite on renome la tmp avec le bon nom et si tous ce passe bien on suprime l'ancienne collection.

ps: quand on set tmpCollection ça change quellque chose dans get collection regarde un peut plus bas